File: 수학(구현)/유클리드 호제법(GCD)/백준_2436_공약수/공약수.cs
using System.Collections;
using static System.Console;
using System;

class Program
{
    static int GCD(int a, int b)
    {
        int p = Math.Max(a, b);
        int q = Math.Min(a, b);
        int left = p % q;
        while (left != 0)
        {
            p = q;
            q = left;
            left = p % q;
        }
        return q;
    }
    public static void Main()
    {
        string[] s = ReadLine().Split(' ');
        int g = int.Parse(s[0]);
        int l = int.Parse(s[1]);

        if (g == l) Write("{0} {1}", g, g);
        else
        {
            int p = l / g;
            int sum = p + 2;
            int a = 0, b = 0;

            for (int i = 1; i < p; i++)
            {
                if (p % i == 0)
                {
                    int j = p / i;
                    if (GCD(i, j) == 1)
                    {
                        if (i + j < sum)
                        {
                            sum = i + j;
                            a = i; b = j;
                        }
                        else break;
                    }
                }
            }
            if (a < b) Write("{0} {1}", a * g, b * g);
            else Write("{0} {1}", b * g, a * g);

        }
    }
}

File: 수학(구현)/유클리드 호제법(GCD)/백준_2436_공약수/공약수.py
def GCD(a,b):
    p = max(a,b)
    q = min(a,b)
    left = p%q
    while left != 0:
        p = q
        q = left
        left = p % q
    return q

g,l = map(int, input().split(' '))
if g == l:
    print(f'{g} {g}')
else:
    p = l//g
    sum = p+2
    a = 0
    b = 0
    for i in range(1, p):
        if p % i == 0:
            j = p // i
            if GCD(i, j) == 1:
                if i + j < sum:
                    sum = i + j
                    a = i
                    b = j
                else:
                    break
    if a < b:
        print(f'{a*g} {b*g}')
    else:
        print(f'{b*g} {a*g}')